### PR TITLE
Edit Profile fix

### PR DESCRIPTION
**Pull request by Starhkz**
This fix also includes updates to the profile model, a service to get a user's profile and an addition to the storage keys.

### CODE REVIEW

File: lib/ui/view/edit_profile/edit_profile_viewmodel.dart [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
-String _nameD = '', _displayD = '', _statusD = '', _phoneD = '';
 String _name = '', _display = '', _status = '', _phone = '';
 
-class EditProfileViewModel extends BaseViewModel {
+class EditProfileViewModel extends FutureViewModel {
+  ProfileModel userData = ProfileModel();
   final snackbar = locator<SnackbarService>();
   final navigationService = locator<NavigationService>();
   final connectivityService = locator<ConnectivityService>();
   final _api = HttpApiService('https://api.zuri.chat/');
```
Comment (pauleke65) [inline lib/ui/view/edit_profile/edit_profile_viewmodel.dart:20]: Use the api service from api_service.dart

Comment (Starhkz) [reply lib/ui/view/edit_profile/edit_profile_viewmodel.dart:20]: Ok

File: lib/ui/view/edit_profile/edit_profile_viewmodel.dart [hunk 2]
```diff
@@ -36,164 +34,56 @@ class EditProfileViewModel extends BaseViewModel {
...
-     await _api.patch(link2, data: editData2, headers: {
-      'Authorization':
-          'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
-    });
+  @override
+  Future futureToRun() async {
+    setBusy(true);
+    userData = await GetUserProfile().currentUser();
+    setBusy(false);
+    throw UnimplementedError();
```
Comment (pauleke65) [inline lib/ui/view/edit_profile/edit_profile_viewmodel.dart:85]: Why is this throwing and error?

Comment (Starhkz) [reply lib/ui/view/edit_profile/edit_profile_viewmodel.dart:85]: Unused import

**Review by pauleke65 — CHANGES_REQUESTED** [review]
Great job.
Please check requested changes and then run ```flutter analyze``` and make sure no issues are found

File: lib/ui/view/edit_profile/edit_profile_viewmodel.dart (hunk 2 above)
Comment (pauleke65) [reply lib/ui/view/edit_profile/edit_profile_viewmodel.dart:85]: Yes, but now the the function is implemented, so you can remove the ```throw UnimplementerError();``` 
If it gives an error the you have to return a value

File: lib/ui/view/edit_profile/edit_profile_viewmodel.dart [hunk 3]
```diff
@@ -17,7 +19,8 @@ class EditProfileViewModel extends FutureViewModel {
   final snackbar = locator<SnackbarService>();
   final navigationService = locator<NavigationService>();
   final connectivityService = locator<ConnectivityService>();
-  final _api = HttpApiService('https://api.zuri.chat/');
+  final _api = HttpApiService(coreBaseUrl);
```
Comment (pauleke65) [inline lib/ui/view/edit_profile/edit_profile_viewmodel.dart:22]: make this:_
api = locator<ApiService>();

File: lib/ui/view/edit_profile/edit_profile_viewmodel.dart [hunk 4]
```diff
@@ -17,7 +19,8 @@ class EditProfileViewModel extends FutureViewModel {
...
+  final api = ApiService();
```
Comment (pauleke65) [inline lib/ui/view/edit_profile/edit_profile_viewmodel.dart:23]: remove this

**Review by pauleke65 — APPROVED** [review]
LG

**Review by YenumJay — APPROVED** [review]
Looks great.

**Pull request merged**